core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -203,32 +203,46 @@ ManifestReader<F> scanMetrics(ScanMetrics newScanMetrics) {
   }
 
   CloseableIterable<ManifestEntry<F>> entries() {
-    if ((rowFilter != null && rowFilter != Expressions.alwaysTrue())
-        || (partFilter != null && partFilter != Expressions.alwaysTrue())
-        || (partitionSet != null)) {
+    return entries(false /* all entries */);
+  }
+
+  private CloseableIterable<ManifestEntry<F>> entries(boolean onlyLive) {
+    if (hasRowFilter() || hasPartitionFilter() || partitionSet != null) {
```

Comment:
I had to add `hasRowFilter()` and `hasPartitionFilter()` to avoid cyclomatic complexity warnings.

Reply:
Do you need to consider the condition?
```
partitionSet.isEmpty() 
```
How about adding another method `hasPartitionSet`?